### ISSUE TITLE
false positive svelte/no-unused-props when Props extend other ComponentProps

### ISSUE THREAD

**Issue by Theo-Steiner** [sveltejs/eslint-plugin-svelte#1302]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-svelte` are you using?

3.11.0

### What did you do?

<details>
<summary>Configuration</summary>

```
import js from "@eslint/js";
import sveltePlugin from "eslint-plugin-svelte";
import tseslint from "typescript-eslint";
export default tseslint.config(
  js.configs.recommended,
  ...tseslint.configs.recommended,
  ...sveltePlugin.configs["flat/recommended"],
  {
    files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
    languageOptions: {
      parserOptions: {
        parser: tseslint.parser,
        extraFileExtensions: [".svelte"],
        projectService: true
      },
    }
  },
  {
  	rules: {
			'@typescript-eslint/no-unsafe-assignment': 'error',
		}
  }
);
```
</details>

ComponentA.svelte
```svelte
<script lang="ts">
    let {a}: {a: number} = $props();
</script>

<p>{a}</p>
```

MyComponent.svelte ← error occurs here
```svelte
<script lang="ts">
    import type { ComponentProps } from 'svelte';

    import ComponentA from './ComponentA.svelte'

    interface Props extends ComponentProps<typeof ComponentA> {
        b: number;
    }

    let { a, b }: Props = $props()
</script>

<ComponentA {a} />
<p>{b}</p>
```

### What did you expect to happen?

when using all props (the Component's own props & the extended components props), I expect `svelte/no-unused-props` to not be triggered

### What actually happened?

`Index signature is unused. Consider using rest operator (...) to capture remaining properties.
[(svelte/no-unused-props)]` is reported by eslint

### Link to **GitHub Repo** with Minimal Reproducible Example

[Playground](https://eslint-online-playground.netlify.app/#eNqNU01v2zAM/SuCMMBtEdt3Nw067OM2rMCOVQ+KTQf2bEmQ5KCF4f8+ypIcJ+m6OZeIfHwkH8mRGl3mP96+yF5JAcJm5gidBVrQrSl1oyzpuDg8MGoNozsmCH4NYrUl9k0BGckS+qSlMmQitZY9STxPcs/EWdCC/hxwWX4yheTJEiMs6JqXQDw3vFoQlblIuXWFyHpFvSOjJ3DfviBi6PegsRT3niJ7BxbL5xuyJ1MRMjyQT8r9ubllYpt7BbBrfKwKH/lEcrRu1W7cT9tc7egG9SmlqJtD1hopUL65AkYbUXZDBYwW5NmnZbPkd6FXRjeLeTZkVeakdsYXJiZkdvArjT4c0NwZx6ZGHpufVq2hEOe9YR889gGmQ9mzpRlMFEbXGj8yRh89KG8xJZIFvy/sqRsOjYhID0zVbExjy6cYazwi4t0ofWWpd8xgeJ3BFdR86E5Bocgb13RrwstkGkrZ97goUM3aZhkqehbxLmZdfsQ9M1p33OYrOKMvc0TYsLrpwOBsk7u7ZabJhqyfmP3S0prEs+CscHoDP8BPZRspkGpZXcW1AX1tj57iJIQ3BEb34aVo/h1r++ZOxngG7GbZuph+ptOyhdL+An1sSkBaPUD0TgGHVxMffrMRNHeOL2bdL3m8Gl4uZDoIw2tIuTHNQaCENilIAlpLnSCbC/TUTNze4/opXv5GNS6uyDG7E2K0lxXmDVfDaAXHr6DcYETZAK7jotP5lrpQHCQYXKjl4OKGfeC83Nx3of/hDIKkflB/gZ70+ydguY41Lug40ekPPTf3mw==)
